apps/extension/src/routes/popup/index.tsx

```diff
@@ -16,26 +27,52 @@ export const popupIndexLoader = async () => {
   }
 
   const password = await sessionExtStorage.get('hashedPassword');
+
   if (!password) {
     return redirect(PopupPath.ENTER_PASSWORD);
   }
   return null;
 };
 
 export const PopupIndex = () => {
-  const { all } = useStore(accountsSelector);
+  const navigate = usePopupNav();
+  const { clearSessionPassword, hashedPassword } = useStore(passwordSelector);
+
+  useEffect(() => {
+    if (!hashedPassword) navigate(PopupPath.ENTER_PASSWORD);
+  }, [hashedPassword]);
```

Comment:
Remove. See loader above.

Reply:
doesnt redirect when logout

Reply:
Can you add a code comment, that the useEffect() is for that purpose?